rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,572 @@
+- Many existing comments written for documentation will remain imperceptible.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Will `nix` itself implement native support like in rust -> `cargo doc`?
+
+- How can a tool keep the connection from where a docstring was defined and where the attribute was exposed (lib/default. Nix exposes mapAttrs which is defined at lib/attrsets. Nix)
+  - There are more complicated things.
+
+**Possible answer**: This is a tooling question; implementation details will not be discussed in detail. It is possible to track comments in the AST. As shown before in this PR: [nix/#1652](https://github.com/NixOS/nix/pull/1652)
+
+# Future work
+[Future]: #future-work
+
+## Editor support
+
+- Implement displaying the related documentation when hovering over an expression. (lspAction/hover)
+
+Nix already offers a bunch of LSPs, e.g., [nil](https://github.com/oxalica/nil), [rnix-lsp](https://github.com/nix-community/rnix-lsp) are the most common ones.
+
+## Nixdoc
+
+Nixdoc may be changed to differentiate between regular comments and doc-comments.
+There might be an intermediate phase of transition, where the old Syntax and features are supported in parallel to allow a phase of transition and refactoring of existing documentation comments.
+
+## Documentation generators
+
+We think that a future documentation tool could be out of one of the two following categories.
+
+- (1) Tools that utilize static code analysis and configuration files.
+
+- (2) Tools that use dynamic evaluation to solve for tracking down name and value relationships and provide more accurate documentation with less configuration overhead.
+
+  > We could solve such concerns in [`tvix`](https://tvix.dev/) or in `nix`, which could vend a tool that pre-evaluates expressions and gathers their respective documentation.
+
+For the beginning it seems promising to start with the static approach (1). In the long term dynamic approach (2) seems more promising and accurate but requires a much deeper understanding of compilers and evaluators.
+However the border between those two variants is not strict and the community might even build tools that live in between.  
+
+## Type
+
+- An RFC under construction specifies the used Syntax within the `# Type` Heading.
+- The `type` feature should belong in the nix syntax. Try them within the comments first; This is still possible.
+
+- see a [preview](https://typednix.dev) of an eventual future doc-type-syntax.
+
+## Native support in Nix
+
+- `NixOS/nix` could implement native support for doc-comments so that our users do not have to rely on nixpkgs or external tools. Those tools can still exist and provide more custom functionality, but documenting nix expressions should be natively possible.
+
+## References
+
+### Other Conventions
+
+- [Rust](https://doc.rust-lang.org/stable/reference/comments.html#doc-comments)
+- [Python](https://peps.python.org/pep-0257/)
+- [JSDoc](https://jsdoc.app/)
```

Comment:
I would also add https://go.dev/doc/comment, which is an example of an ecosystem not introducing a new comment format.